azurelinuxagent/common/osutil/default.py

```diff
@@ -630,9 +635,11 @@ def umount_dvd(self, chk_err=True, mount_point=None):
 
     def eject_dvd(self, chk_err=True):
         dvd = self.get_dvd_device()
-        retcode = shellutil.run("eject {0}".format(dvd))
-        if chk_err and retcode != 0:
-            raise OSUtilError("Failed to eject dvd: ret={0}".format(retcode))
+        try:
+            shellutil.run_command(["eject", dvd])
+        except shellutil.CommandError as cmd_err:
+            if chk_err:
+                raise OSUtilError("Failed to eject dvd: ret={0}".format(cmd_err.returncode))
```

Comment:
we should probably add the stdout/stderr of the command (the original code was logging errors)